### ISSUE TITLE
haproxy config error while running docker 

### ISSUE THREAD

**Issue by vargha1** [WhatsApp/proxy#9]
Certificate generation completed.
/
[NOTICE]   (21) : haproxy version is 2.6.7-c55bfdb
[ALERT]    (21) : config : parsing [/usr/local/etc/haproxy/haproxy.cfg:57] : fetch method 'ipv4' : failed to parse 'not' as type 'IPv4 address' at position 1
[ALERT]    (21) : config : parsing [/usr/local/etc/haproxy/haproxy.cfg:66] : fetch method 'ipv4' : failed to parse 'not' as type 'IPv4 address' at position 1
[ALERT]    (21) : config : parsing [/usr/local/etc/haproxy/haproxy.cfg:75] : fetch method 'ipv4' : failed to parse 'not' as type 'IPv4 address' at position 1
[ALERT]    (21) : config : Error(s) found in configuration file : /usr/local/etc/haproxy/haproxy.cfg
[ALERT]    (21) : config : Fatal errors found in configuration.

**Comment by slawlor** [comment]
Can you show the full command you're running? Did you set any environment variables prior to starting the container? (like `PUBLIC_IP`)?

**Comment by vargha1** [comment]
docker run -it -p 80:80 -p 443:443 -p 5222:5222 -p 8080:8080 -p 8443:8443 -p 8222:8222 -p 8199:8199 86727b3a6a9b

this is the command.
Where should i set/change My settings like (PUBLIC_IP)

thanks.

**Comment by slawlor** [comment]
Hmm something strange is going on, because it's causing a validation error at lines which are comments in the configuration file. Perhaps it's a file encoding problem? If you could try removing the `#PUBLIC_IP` lines in `proxy/src/proxy_config.cfg` and re-build the image, that might work. We're working on publishing a common image now for public consumption which will remove the need for people to build it manually (hopefully avoiding these issues)

**Comment by AliSepehri** [comment]
@slawlor Removing `#PUBLIC_IP` from `proxy/src/proxy_config.cfg` file and rebuilding the docker image, worked for me.

**Comment by slawlor** [comment]
@AliSepehri Thanks, I'll look into if there's encoding problems and investigate. For know feel free to use it without the public ip statements in the configuration

**Comment by ahmadmousavian** [comment]
نیاز به پروکس

**Comment by ahmadmousavian** [comment]
need proxy

**Comment by nimaredrose** [comment]
DigitalOcean has same error. is there any solution ?

**Comment by arash7444** [comment]
Hi again, 
I have similar issue (snapshot below). Could you please explain more what meant by 

> removing the #PUBLIC_IP lines in proxy/src/proxy_config.cfg

what do I need to remove from thisfile?
![image](https://user-images.githubusercontent.com/21036995/211922863-4391f985-6730-4d3f-bf0e-f629b0db28ef.png)



![image](https://user-images.githubusercontent.com/21036995/211922319-9dc9544d-88b8-4d35-8a43-e27c9b309034.png)

**Comment by Ktechengr** [comment]
Sorry i am examining the file now please give me some time to see

**Comment by Ktechengr** [comment]
Bind ipv4 // 80
                  //8080

Clear public port from configuration

**Comment by arash7444** [comment]
Thanks for the help, but it did not work :-(

**Comment by iranazad2023** [comment]
> DigitalOcean has the same error. is there any solution?

Yes, as mentioned, you have to remove #PUBLIC IP from the proxy_config.cfg. The file is located at .../proxy/src. If you are not familiar with Linux, follow the below command in your terminal. Before you manually execute the container, go to the src directory as follow:

cd proxy
cd src
nano proxy_config.cfg 
So the last one would open the file, then you have to erase three #Public IP in your file, then press Esc and save the changes. SO go back to the main directory with the following:
cd ...
cd ...
So now run again:
docker run -it -p 80:80 -p 443:443 -p 5222:5222 -p 8080:8080 -p 8443:8443 -p 8222:8222 -p 8199:8199 whatsapp_proxy:1.0

good luck!

**Comment by amirarshia** [comment]
I have done this but still get the same error. 
![image](https://user-images.githubusercontent.com/37936213/212780202-0c135a51-a6ba-40a0-944d-c0227e7ecd43.png)
![image](https://user-images.githubusercontent.com/37936213/212780240-4980d1ad-b2be-47b3-8c64-fa2a1b9b791a.png)

**Comment by iranazad2023** [comment]
> I have done this but still get the same error.

So If you are still facing the same error, it is better after edit the proxy file you only  use one (cd ..) and run again it at the proxy directory instead. You should solve it this time.

**Comment by shsedghi** [comment]
@amirarshia When you modified proxy_config.cfg do the build process again to build the docker container using `docker build proxy/ -t whatsapp_proxy:1.0` before `docker run -it -p 80:80 -p 443:443 -p 5222:5222 -p 8080:8080 -p 8443:8443 -p 8222:8222 -p 8199:8199 whatsapp_proxy:1.0`
